test/abstractarray.jl

```diff
@@ -340,91 +356,109 @@ function test_scalar_indexing{T}(::Type{T}, shape, ::Type{TestAbstractArray})
         end
     end
     @test C == B == A
-    # Test zero-dimensional setindex
-    A[] = 0; B[] = 0
-    @test A[] == B[] == 0
-    @test A == B
+    @testset  "Test zero-dimensional setindex" begin
+        A[] = 0; B[] = 0
+        @test A[] == B[] == 0
+        @test A == B
+    end
 end
 
 function test_vector_indexing{T}(::Type{T}, shape, ::Type{TestAbstractArray})
-    N = prod(shape)
-    A = reshape(collect(1:N), shape)
-    B = T(A)
-    idxs = rand(1:N, 3, 3, 3)
-    @test B[idxs] == A[idxs] == idxs
-    @test B[vec(idxs)] == A[vec(idxs)] == vec(idxs)
-    @test B[:] == A[:] == collect(1:N)
-    @test B[1:end] == A[1:end] == collect(1:N)
-    @test B[:,:] == A[:,:] == reshape(1:N, shape[1], prod(shape[2:end]))
-    @test B[1:end,1:end] == A[1:end,1:end] == reshape(1:N, shape[1], prod(shape[2:end]))
-    # Test with containers that aren't Int[]
-    @test B[[]] == A[[]] == []
-    @test B[convert(Array{Any}, idxs)] == A[convert(Array{Any}, idxs)] == idxs
-
-    # Test adding dimensions with matrices
-    idx1 = rand(1:size(A, 1), 3)
-    idx2 = rand(1:Base.trailingsize(A, 2), 4, 5)
-    @test B[idx1, idx2] == A[idx1, idx2] == reshape(A[idx1, vec(idx2)], 3, 4, 5) == reshape(B[idx1, vec(idx2)], 3, 4, 5)
-    @test B[1, idx2] == A[1, idx2] == reshape(A[1, vec(idx2)], 4, 5) == reshape(B[1, vec(idx2)], 4, 5)
-
-    # test removing dimensions with 0-d arrays
-    idx0 = reshape([rand(1:size(A, 1))])
-    @test B[idx0, idx2] == A[idx0, idx2] == reshape(A[idx0[], vec(idx2)], 4, 5) == reshape(B[idx0[], vec(idx2)], 4, 5)
-    @test B[reshape([end]), reshape([end])] == A[reshape([end]), reshape([end])] == reshape([A[end,end]]) == reshape([B[end,end]])
-
-    # test logical indexing
-    mask = bitrand(shape)
-    @test B[mask] == A[mask] == B[find(mask)] == A[find(mask)] == find(mask)
-    @test B[vec(mask)] == A[vec(mask)] == find(mask)
-    mask1 = bitrand(size(A, 1))
-    mask2 = bitrand(Base.trailingsize(A, 2))
-    @test B[mask1, mask2] == A[mask1, mask2] == B[find(mask1), find(mask2)]
-    @test B[mask1, 1] == A[mask1, 1] == find(mask1)
+    @testset "test_vector_indexing{T}" begin
+        N = prod(shape)
+        A = reshape(collect(1:N), shape)
+        B = T(A)
+        idxs = rand(1:N, 3, 3, 3)
+        @test B[idxs] == A[idxs] == idxs
+        @test B[vec(idxs)] == A[vec(idxs)] == vec(idxs)
+        @test B[:] == A[:] == collect(1:N)
+        @test B[1:end] == A[1:end] == collect(1:N)
+        @test B[:,:] == A[:,:] == reshape(1:N, shape[1], prod(shape[2:end]))
+        @test B[1:end,1:end] == A[1:end,1:end] == reshape(1:N, shape[1], prod(shape[2:end]))
+
+        @testset "Test with containers that aren't Int[]" begin
```

Comment:
@tkelman Maybe I should also avoid using testsets for comments inside functions like I've done here?